service/matching/matcher_test.go

```diff
@@ -295,7 +302,7 @@ func (t *MatcherTestSuite) TestForwardingWhenBacklogIsEmpty() {
 		gomock.Any(),
 	).Return(&matchingservice.PollWorkflowTaskQueueResponse{}, errMatchingHostThrottleTest)
 	_, e := t.matcher.Poll(ctx, &pollMetadata{})
-	t.ErrorAs(e, &errNoTasks)
+	t.ErrorAs(e, &targetErr)
```

Comment:
This is the actual flake fix

Reply:
Shouldn't this be `t.ErrorIs(e, errNoTasks)`? `errNoTasks` isn't a type that implements error, it's just a specific value (from `errors.New`).  This is equivalent to `t.Error(e)`

Reply:
Actually I see a lot of calls in this file to `t.Error(err, something)` that should probably be `t.ErrorIs(err, something)`. `t.Error` only checks that an error is non-nil, and the something is the message.

Reply:
Thank you. I pulled this change from another change set and didn't even think to ask myself that.